Analysis/Stan/mapping_model_generate.stan

```diff
@@ -75,14 +77,17 @@ data {
   int <lower=0, upper=1> censored[K1]; // Whether data is censored in the mapping from observations to location/times
   
   // Mappings
-  int <lower=0, upper=M> map_obs_loctime_obs[K1];    // the observation side of the mapping from observations to location/times
-  int <lower=0, upper=L> map_obs_loctime_loc[K1];    // the location side of the mapping from observations to location/times
-  int <lower=0, upper=L> map_loc_grid_loc[K2];       // the location side of the mapping from locations to gridcells
-  int <lower=0, upper=N> map_loc_grid_grid[K2];      // the gridcell side of the mapping from locations to gridcells
-  int <lower=0, upper=smooth_grid_N> map_smooth_grid[N];    // vector with repeating smooth_grid_N indexes repeating 1:N
+  int <lower=1, upper=M> map_obs_loctime_obs[K1];    // the observation side of the mapping from observations to location/times
+  int <lower=1, upper=L> map_obs_loctime_loc[K1];    // the location side of the mapping from observations to location/times
+  int <lower=1, upper=L> map_loc_grid_loc[K2];       // the location side of the mapping from locations to gridcells
+  int <lower=1, upper=N> map_loc_grid_grid[K2];      // the gridcell side of the mapping from locations to gridcells
+  int <lower=1, upper=smooth_grid_N> map_smooth_grid[N];    // vector with repeating smooth_grid_N indexes repeating 1:N
   real <lower=0, upper=1> map_loc_grid_sfrac[K2];    // the population-weighed location spatial fraction covered by each gridcell
   int <lower=1, upper=N_space> map_spacetime_space_grid[N];  // map from spacextime grid cells to space-only grid
   int<lower=1, upper=N_admin_lev> map_obs_admin_lev[M];    // administrative level of each observation for observation model
+  int <lower=1, upper=L> map_loctime_combs_loc[K3];             // the location time side of the mapping from locations to unique location time combinations
+  int <lower=1, upper=L_combs> map_loctime_combs_comb[K3];      // the unique combination side of the mapping from locations to unique location time combinations
+  int <lower=1, upper=N_admin_lev> map_u_loctime_combs_admin_lev[L_combs];      // the unique combination side of the mapping from locations to unique location time combinations
```

Comment:
For  map_u_loctime_combs_admin_lev, couldn't there be an admin level 0? A bit confused though because the comment says this is the unique combination side but the upper limit is N_admin_lev. 
 
Also, the variable name is getting long, but shouldn't there be a "admin" or "comb" at the end of the variable name to follow the naming convention of all of the other map objects (Depending on which side of the mapping it actually is)

Reply:
For the admin levels there are indexed based on a vector of unique admin levels present in the data, so here it starts at 1 and goes up to N_admin_levels. This is to allow for possibly missing admin levels in the data (for instance 0, 2 and 3).

Yes I agree that the naming convention is not followed here, it is more following the way we named the map of obs to admin levels. The reason we can accept this I think is that we will never be doing the mapping in the other direction.